raiden/raiden_event_handler.py

```diff
@@ -603,104 +611,215 @@ def handle_contract_send_channelunlock(
+        else:
+            message = SettlementRequiredLightMessage(
+                channel_identifier=channel_settle_light_event.channel_identifier,
+                channel_network_identifier=channel_settle_light_event.token_network_identifier,
                 participant1=payment_channel.participant1,
+                participant1_transferred_amount=settlement_parameters.transferred_amount,
+                participant1_locked_amount=settlement_parameters.locked_amount,
+                participant1_locksroot=settlement_parameters.locksroot,
                 participant2=payment_channel.participant2,
-                block_identifier=triggered_by_block_hash,
-                channel_identifier=channel_settle_event.channel_identifier,
+                participant2_transferred_amount=settlement_parameters.partner_transferred_amount,
+                participant2_locked_amount=settlement_parameters.partner_locked_amount,
+                participant2_locksroot=settlement_parameters.partner_locksroot)
+
+        log.debug("Storing light client message to require settle")
+
+        message_already_stored = LightClientMessageHandler.is_message_already_stored(
+            light_client_address=payment_channel.participant1,
+            message_type=LightClientProtocolMessageType.SettlementRequired,
+            unsigned_message=message,
+            wal=raiden.wal)
+
+        if message_already_stored:
+            log.debug(
+                "Skipping storing light client settle message "
+                "for {} with type {} since already exists in database".format(
+                    payment_channel.participant1.hex(),
+                    str(LightClientProtocolMessageType.SettlementRequired)))
+        else:
+            LightClientMessageHandler \
+                .store_light_client_protocol_message(identifier=message_identifier,
+                                                     message=message,
+                                                     signed=False,
+                                                     payment_id=None,
+                                                     light_client_address=payment_channel.participant1,
+                                                     order=0,
+                                                     message_type=LightClientProtocolMessageType.SettlementRequired,
+                                                     wal=raiden.wal)
+
+    @staticmethod
+    def process_data_and_get_settlement_parameters(raiden: "RaidenService",
+                                                   token_network_identifier: TokenNetworkID,
+                                                   channel_identifier: ChannelID,
+                                                   triggered_by_block_hash: BlockHash) -> SettlementParameters:
+
+        log.debug("Processing settlement data")
+
+        assert raiden.wal, "The Raiden Service must be initialize to handle events"
```

Comment:
typo in `initialized`

Reply:
This was a fragment from somewhere else, good catch we should fix it all over the place

Reply:
Done